pkg/kvstore/etcd_test.go

```diff
@@ -14,13 +14,91 @@ import (
 	"testing"
 	"time"
 
+	"github.com/cilium/cilium/pkg/testutils"
 	"github.com/stretchr/testify/require"
 	etcdAPI "go.etcd.io/etcd/client/v3"
 	"k8s.io/apimachinery/pkg/util/rand"
-
-	"github.com/cilium/cilium/pkg/testutils"
 )
 
+// SetupDummyWithConfigOpts sets up the dummy kvstore for tests but also
+// configures the module with the provided opts. A lock mechanism it used to
+// prevent the creation of two clients at the same time, to avoid interferences
+// in case different tests are run in parallel. A cleanup function is
+// automatically registered to delete all keys and close the client when the
+// test terminates.
+func SetupDummyWithConfigOpts(tb testing.TB, dummyBackend string, opts map[string]string) {
```

Comment:
This is duplicating a non-trivial function in two different places, which will eventually diverge. I'd suggest trying to refactor the structure to avoid requiring this duplication.